### PR TITLE
[FEAT] Store Audio Settings in Local Storage

### PR DESCRIPTION
**Pull request by c-py**
# Description

Refreshing the browser resets the audio volume and game sound settings. 

This change updates the audio player to store configured audio settings in browser local storage. This allows the settings to persist across browser refresh.

The changes introduce a new react hook, `useLocalStorage` to replace the current usage of `useState` in the audio player.

## Changes

- Introduces a new hook, `useLocalStorage`
- Updates the `AudioPlayer.tsx` hooks from `useState` to `useLocalStorage`.

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tests have been created in `useLocalStorage.test.ts`. The tests cover use cases for storing strings, booleans, numbers and JSON objects.

# Checklist:

- [X] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [X] Screenshot if it includes any UI changes
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/ui/AudioPlayer.tsx [hunk 1]
```diff
@@ -11,13 +11,25 @@ import { Button } from "components/ui/Button";
...
+  );
+  const [isPlaying, setPlaying] = useLocalStorage<boolean>(
+    "audioplayer.playing",
+    true
+  );
+  const [lsVolume, setLsVolume] = useLocalStorage<number>(
+    "audioplayer.volume",
+    0.1
+  );
+  const volume = useStepper({ initial: lsVolume, step: 0.1, max: 1, min: 0 });
```
Comment (c-py) [inline src/components/ui/AudioPlayer.tsx:31]: Passing in `lsVolume` here results in an unnecessary rerender each time the volume stepper is changed.

Any thoughts to resolve this cleanly are welcome.

File: src/lib/utils/hooks/__tests__/useLocalStorage.test.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    expect(value).toBe(defaultValue);
+  });
+  it("starts with a default object", () => {
+    const defaultValue = { defaultKey: ["1", 2] };
+
+    const { result } = renderHook(() => useLocalStorage(key, defaultValue));
+    const [value] = result.current;
+
+    expect(value).toStrictEqual(defaultValue);
+  });
```
Comment (c-py) [inline src/lib/utils/hooks/__tests__/useLocalStorage.test.ts:55]: This might be a bit redundant. Happy to remove if this is testing typescript too much.

**Comment by Pranav-yadav** [comment]
It's nice that you've started contributing, but you should've looked the open issue first before starting on this.
This is the open issue on the in-game sounds and settings https://github.com/sunflower-land/sunflower-land/issues/547

Specifically I've mentioned about the feature is **implemented**, here https://github.com/sunflower-land/sunflower-land/issues/547#issuecomment-1099454905
If PR for any feat is not opened it doesn't mean the feat is not implemented. Most probably that dev must be testing that feat.

Why waste your as well as someone else's time on same thing.
While contributing to the Open Source projects we need to be extra careful about these things.

Anyway I'm pushing those changes.

Please don't think that I'm rude, Adam and other maintainers will also suggest the same. 👍 
Btw, Welcome to the SFL GitHub community. 😃

**Comment by c-py** [comment]
> It's nice that you've started contributing, but you should've looked the open issue first before starting on this. This is the open issue on the in-game sounds and settings #547
> 
> Specifically I've mentioned about the feature is **implemented**, here [#547 (comment)](https://github.com/sunflower-land/sunflower-land/issues/547#issuecomment-1099454905) If PR for any feat is not opened it doesn't mean the feat is not implemented. Most probably that dev must be testing that feat.
> 
> Why waste your as well as someone else's time on same thing. While contributing to the Open Source projects we need to be extra careful about these things.
> 
> Anyway I'm pushing those changes.
> 
> Please don't think that I'm rude, Adam and other maintainers will also suggest the same. 👍 Btw, Welcome to the SFL GitHub community. 😃

@Pranav-yadav No problem at all, thanks for pointing it out! And thanks for welcome - it's good to be here 🎉 !

I checked the open issues, but didn't realise that they included persisting the settings between refreshes. Happy to close this one off if the changes have already been implemented.

**Comment by Pranav-yadav** [comment]
> @Pranav-yadav No problem at all, thanks for pointing it out! And thanks for welcome - it's good to be here 🎉 !
> 
> I checked the open issues, but didn't realise that they included persisting the settings between refreshes. Happy to close this one off if the changes have already been implemented.

Glad to hear that. 😄 
Opened PR #659

**Comment by c-py** [comment]
> > @Pranav-yadav No problem at all, thanks for pointing it out! And thanks for welcome - it's good to be here 🎉 !
> > I checked the open issues, but didn't realise that they included persisting the settings between refreshes. Happy to close this one off if the changes have already been implemented.
> 
> Glad to hear that. 😄 Opened PR #659

Nice work!

I'll go ahead and close this one - looks like you have the persistence well and truly covered.

**Pull request closed without merging**